db/src/tx_observer.rs

```diff
@@ -174,42 +154,28 @@ impl Drop for TxObservationService {
     }
 }
 
-pub struct InProgressObserverTransactWatcher<'a> {
-    collected_datoms: AttributeSet,
-    observer_service: &'a Mutex<TxObservationService>,
-    active: bool
+pub struct InProgressObserverTransactWatcher {
+    collected_attributes: AttributeSet,
```

Comment:
I called it `collected_datoms` mostly because the next stage will be to collect more than just attributes for more advanced observer filtering and there didn't seem much point renaming it, but this does make it more descriptive for the immediacy.